src/FEValues/common_values.jl

```diff
@@ -268,16 +275,16 @@ function spatial_coordinate(fe_v::AbstractValues, q_point::Int, x::AbstractVecto
     return vec
 end
 
-function Base.show(io::IO, ::MIME"text/plain", fe_v::AbstractValues)
-    print(io, "$(typeof(fe_v)) with $(getnbasefunctions(fe_v)) shape functions and $(getnquadpoints(fe_v)) quadrature points")
-end
 
-# copy
-for ValueType in (CellValues, FaceValues)
-    args = [:(copy(cv.$fname)) for fname in fieldnames(ValueType)]
-    @eval begin
-        function Base.copy(cv::$ValueType)
-            return typeof(cv)($(args...))
-        end
+function shape_gradients_and_values!(gradients::AbstractMatrix, values::AbstractMatrix, ip, qr::QuadratureRule)
+    for (qp, ξ) in pairs(getpoints(qr))
+        shape_gradients_and_values!(@view(gradients[:, qp]), @view(values[:, qp]), ip, ξ)
+    end
+end
+#= PR798
+function shape_hessians_gradients_and_values!(hessians::AbstractMatrix, gradients::AbstractMatrix, values::AbstractMatrix, ip, qr::QuadratureRule)
+    for (qp, ξ) in pairs(getpoints(qr))
+        shape_hessians_gradients_and_values!(@view(hessians[:, qp]), @view(gradients[:, qp]), @view(values[:, qp]), ip, ξ)
     end
 end
+=#
```

Comment:
```suggestion
=#

```

Reply:
I guess because of Windows I have a line here but not online...